File: libgomp/testsuite/libgomp.oacc-fortran/pr70828-2.f90
! Subarrays declared on data construct: assumed-shape array.

subroutine s1(n, arr)
  integer :: n
  integer :: arr(n)

  !$acc data copy(arr(5:n-10))
  !$acc parallel loop
  do i = 10, n - 10
     arr(i) = i
  end do
  !$acc end parallel loop
  !$acc end data
end subroutine s1

program test
  integer, parameter :: n = 100
  integer i, data(n)

  data(:) = 0

  call s1(n, data)

  do i = 1, n
     if ((i < 10 .or. i > n-10)) then
        if ((data(i) .ne. 0)) call abort
     else if (data(i) .ne. i) then
        call abort
     end if
  end do
end program test

File: libgomp/testsuite/libgomp.oacc-c-c++-common/pr70828-2.c
/* Subarray declared on data construct, accessed through pointer.  */

#include <assert.h>

void
s1 (int *arr, int c)
{
#pragma acc data copy(arr[5:c-10])
  {
#pragma acc parallel loop
    for (int i = 5; i < c - 5; i++)
      arr[i] = i;
  }
}

int
main (int argc, char* argv[])
{
  const int c = 100;
  int arr[c];

  for (int i = 0; i < c; i++)
    arr[i] = 0;

  s1 (arr, c);

  for (int i = 0; i < c; i++)
    if (i >= 5 && i < c - 5)
      assert (arr[i] == i);
    else
      assert (arr[i] == 0);

  return 0;
}
